packages/network-controller/src/rpc-service/rpc-service.ts

```diff
@@ -250,7 +251,8 @@ export class RpcService implements AbstractRpcService {
           // Ignore server sent HTML error pages or truncated JSON responses
           error.message.includes('not valid JSON') ||
           // Ignore server overload errors
-          error.message.includes('Gateway timeout') ||
+          ('httpStatus' in error &&
+            (error.httpStatus === 503 || error.httpStatus === 504)) ||
```

Comment:
The old middleware used to consider 503/504 retriable only if they included the message "Gateway timeout".

When that logic was replaced with the RPC service, it was widened to apply to all 503/504 responses for some reason.

I'm preserving that behavior here, as it seems fairly harmless to widen the set of responses we retry, and because I'm not sure why it was done.